lib/graceful-exit.js

```diff
@@ -148,8 +150,22 @@ exports.middleware = function middleware (app) {
 
   return function checkIfExitingGracefully (req, res, next) {
     if (app.settings.graceful_exit === true) {
-      // sorry keep-alive connections, but we need to part ways
-      req.connection.setTimeout(1);
+      if (options.respondDuringExit) {
+        logger('Server unavailable, incoming request rejected with error: ' + req && req.headers || '?');
+        // Signal the caller that the connection is closing, if before suicideTimeout
+        // 2019-09-10 - HH - Fix request processing after connection closes, see issue 14
+        res.set('Connection', 'close');
+        return next(
```

Comment:
I think we need to add one more option for `runLastRequestPerConnectionOnExit` - That, combined with `respondDuringExit` would do what I had done in #15: dispatch the last incoming request received on any still-open connection, but respond with a Connection: close.  That is currently the only combination that I know for sure works in my environment and I'm positive that there will be other users that need that exact behavior as well.  If we can add that extra flag and have it just call next() without error when set then I think we'd be 110% feature complete!  :)

Reply:
:thinking:  Sure, and that follows the standard use of the connection close header as well. Will do.

Reply:
@huntharo - I've added a `performLastRequest` option to that effect, improved names for other options and improved code clarity in the exit handler

Reply:
Looks great now, thanks for that!